pkg/pmem-csi-operator/utils/utils.go

```diff
@@ -34,7 +36,30 @@ const (
 	defaultOperatorNamespace = metav1.NamespaceSystem
 )
 
-func GetKubernetesVersion() (*version.Info, error) {
+var (
+	k8sVersion  uint64
+	versionLock sync.Mutex
```

Comment:
Package-level variables are generally considered bad style (https://dave.cheney.net/2020/02/23/the-zen-of-go, "Avoid package level state"). Is that really necessary here? It's going to be simpler to cache the Kubernetes version where it is needed multiple times.